dds/idl/annotations.h

```diff
@@ -222,6 +273,67 @@ class HashidAnnotation : public AnnotationWithValue<std::string> {
 public:
   std::string definition() const;
   std::string name() const;
+
+private:
+  virtual const std::string* default_value() const
+  {
+    static std::string v = "";
+    return &v;
+  }
+};
+
+// @optional ===================================================================
+
+class OptionalAnnotation : public AnnotationWithValue<bool>, public AbsentValue<bool> {
+public:
+  OptionalAnnotation()
+    : AbsentValue(false)
+  {}
+
+  std::string definition() const;
+  std::string name() const;
+
+  virtual const bool* default_value() const
+  {
+    static bool v = true;
+    return &v;
+  }
```

Comment:
I don't understand what these `default_value` functions are supposed to be doing. If the annotation is there it has the value, if it's not then you use what you're calling `absent_value` (what I think used to be my `default_value`).

Reply:
See comment about default_value above.  I wrote the code assume that annotations with default values would not have a parameter if no value was supplied.  Thus, we had to provide a default value.  Maybe this was just poor naming.  Thus, default value should return the default value as defined in the annotation.

Reply:
> See comment about default_value above. I wrote the code assume that annotations with default values would not have a parameter if no value was supplied. Thus, we had to provide a default value.

Okay that explains quite a bit. No that's not the case.

> Maybe this was just poor naming. Thus, default value should return the default value as defined in the annotation.

Why would you want that? Or is this part of the misunderstanding?

If it helps I think absent_value is a much better name for my `default_value`, since it can't be confused with the default values of the annotations themselves.